bus/src/main/java/org/hawkular/metrics/component/publish/InsertedDataSubscriber.java

```diff
@@ -64,8 +68,21 @@
 
     private Subscription subscription;
 
+    @Inject
+    private PublishCommandTable publish;
+
+    @Inject
+    @Configurable
+    @ConfigurationProperty(METRICS_PUBLISH_PERIOD)
+    private String publishPeriodProperty;
+    private int publishPeriod;
+
     public void onMetricsServiceReady(@Observes @ServiceReady ServiceReadyEvent event) {
-        Observable<List<Metric<?>>> events = event.getInsertedData().buffer(50, TimeUnit.MILLISECONDS, 100)
+        publishPeriod = getPublishPeriod();
+        Observable<List<Metric<?>>> events = event.getInsertedData()
+                .filter(m -> m.getType() != MetricType.STRING)
+                .filter(m -> publish.isPublished(m.getMetricId()))
```

Comment:
OK, I see what you are doing, you are stripping the HMetrics prefix on the alerting side, when putting the metricId in the cache.  Ignoring tenantId and type is likely fine for filtering, at worst we process some unnecessary metrics in Alerting.

It's unfortunate that Hmetrics decided to make metric uniqueness involve type.  I think it may have been better if they assigned a type to a metric but still disallowed duplicate metricIds within a tenant.  Now there really is no choice but to ensure that Halerting dataIds for Hmetrics data are type-prefixed.  I think it could be useful to change the prefixes to be something more specific to Hmetrics, so that there can be no confusion between dataIds defined for Hmetrics metrics and dataIds for data being fed from other sources.  Instead of: AVAILABILITY, GAUGE, etc, perhaps HM_A_, HM_G, etc.  Something more unique.  All clients creating triggers for Hmetrics metrics will need to follow the dataId naming convention.